include/annotated.hpp

```diff
@@ -92,6 +93,74 @@ using DefaultSupportedTypes = SupportedTypes<>;
 static_assert(fair::meta::is_instantiation_of<DefaultSupportedTypes, SupportedTypes>);
 static_assert(fair::meta::is_instantiation_of<SupportedTypes<float, double>, SupportedTypes>);
 
+/**
+ * @brief Represents limits and optional validation for an Annotated<..> type.
+ *
+ * The `Limits` structure defines lower and upper bounds for a value of type `T`.
+ * Additionally, it allows for an optional custom validation function to be provided.
+ * This function should take a value of type `T` and return a `bool`, indicating
+ * whether the value passes the custom validation or not.
+ *
+ * Example:
+ * ```
+ * Annotated<float, "example float", Visible, Limits<0.f, 1024.f>>             exampleVar1;
+ * // or:
+ * constexpr auto isPowerOfTwo = [](const int &val) { return val > 0 && (val & (val - 1)) == 0; };
+ * Annotated<float, "example float", Visible, Limits<0.f, 1024.f, isPowerOfTwo>> exampleVar2;
+ * // or:
+ * Annotated<float, "example float", Visible, Limits<0.f, 1024.f, [](const int &val) { return val > 0 && (val & (val - 1)) == 0; }>> exampleVar2;
+ * ```
+ */
+template<auto LowerLimit, decltype(LowerLimit) UpperLimit, auto Validator = nullptr>
+    requires(requires(decltype(Validator) f, decltype(LowerLimit) v) {
+        { f(v) } -> std::same_as<bool>;
+    } || Validator == nullptr)
+struct Limits {
+    using ValueType = decltype(LowerLimit);
+    static constexpr ValueType MinRange = LowerLimit;
+    static constexpr ValueType MaxRange = UpperLimit;
+    static constexpr decltype(Validator) ValidatorFunc = Validator;
+
+    static constexpr bool
+    validate(const ValueType &value) noexcept {
+        if (LowerLimit == UpperLimit) {
```

Comment:
If the limits are equal, they are not applied? Should the value be checked `value == LoweLimit`?

Reply:
yes, the limits are ignored if lower equals the upper limit.

Reply:
The most common case is that parameters are checked for min/max but while doing the unit-tests I wanted to add also the possibility to have only a validator function.

Reply:
I agree that it's a bit suprising to have min == max accept all values instead of the one exact value, but on the other hand having a setting that only accepts a single value is kind of pointless.

The more natural choice imo would be something like
``` c++
template <typename T>
using EmptyLimits = Limits<std::numeric_limits<T>::min, std::numeric_limits<T>::max>;
```
though I'm not sure if this would cause other problems.

I'd be ok with the currently implemented choice, just has to be documented accordingly.